easybuild/easyblocks/m/modeller.py

```diff
@@ -47,57 +97,65 @@ def build_step(self):
 
     def install_step(self):
         """Interactive install of Modeller."""
-
+        python_looseversion = LooseVersion(get_software_version('Python'))
+        if self.loosever < LooseVersion('9.10') and python_looseversion >= LooseVersion('3'):
+            raise EasyBuildError("Modeller version < 9.10 does not support Python3")
         if self.cfg['key'] is None:
             raise EasyBuildError("Easyconfig parameter 'key' is not defined")
-
         cmd = "%s/Install" % self.cfg['start_dir']
-
-        # by default modeller tries to install to $HOME/bin/modeller9.13
-        # get this path to use it in the question/answer
-        default_install_path = os.path.join(os.path.expanduser('~'), 'bin', 'modeller%s' % self.cfg['version'])
-
-        qa = {
-            # installer will autodetect the right arch. [3] = x86_64
-            'Select the type of your computer from the list above [3]:': '',
-            "[%s]:" % default_install_path: self.installdir,
-            'http://salilab.org/modeller/registration.html:': self.cfg["key"],
-            'https://salilab.org/modeller/registration.html:': self.cfg["key"],
-            'Press <Enter> to begin the installation:': '',
-            'Press <Enter> to continue:': ''
-        }
-
-        run_cmd_qa(cmd, qa, log_all=True, simple=True)
+        run_cmd_qa(cmd, self.qa, log_all=True, simple=True)
+        # Determine lib/arch_ according to modeller's architecure naming scheme. After running the installer for the
+        # first time, there should only one subdirectory in lib, e.g. x86_64-intel8. Save this value for later multi_dep
+        # interations, as lib will already be populated.
+        if not self.arch_:
+            self.arch_ = os.listdir(os.path.join(self.installdir, 'lib'))[0]
+        # _modeller.so is provided for different Python versions, namely 2.5, 3.0, 3.2, and >=3.3
+        # We link the _modeller.so and %(installdir)s/modlib into %(installdir)s/lib/python%(pyshortver)s/site-packages
+        # in order to allow multi_deps python
+        py_api_dirname = 'python2.5'
+        pyshortver = '.'.join(get_software_version('Python').split('.')[:2])
+        python_path = os.path.join(self.installdir, 'lib', 'python%s', 'site-packages') % pyshortver
+        if python_looseversion >= LooseVersion('3.0'):
+            py_api_dirname = 'python3.0'
+        if python_looseversion >= LooseVersion('3.2'):
+            py_api_dirname = 'python3.2'
+        if python_looseversion >= LooseVersion('3.3'):
+            py_api_dirname = 'python3.3'
```

Comment:
what about python versions >3.3? does py_api_dirname stay at python3.3?
or should we do `py_api_dirname = 'python%s' % pyshortver` to make it future-proof?

Reply:
@smoors up to Modeller 10.1 only support for api version 2.5, 3.0,3.2, and 3.3 are provided.
It is probably cleaner to adapt the easyblock, if  newer Modeller versions support other python APIs.
I did not check what is the case for the recently released verson 10.2, yet.
I would suggest not to merge this PR as long as it is not tested with the latest Modeller version.

Reply:
@smoors  This is still the case for Modeller 10.2. I added a comment in the EB.
I also updated PR [#12875](https://github.com/easybuilders/easybuild-easyconfigs/pull/12875) to version 10.2